src/zarr/abc/codec.py

```diff
@@ -44,6 +87,20 @@ async def decode(
     ) -> np.ndarray:
         pass
 
+    async def decode_batch(
+        self,
+        chunk_arrays_and_specs: Iterable[tuple[np.ndarray | None, ArraySpec]],
+        runtime_configuration: RuntimeConfiguration,
+    ) -> Iterable[np.ndarray | None]:
```

Comment:
What would be the problem with making `decode` just a special case of `decode_batch`, where the length of `chunk_arrays_and_specs` is 1? 

Also, it seems odd that `decode_batch` with a batch size of 1 could be nullable, while `decode` with the same argument would not be.

Reply:
That would be possible. If we want to make it easier for the community to develop codecs, the batch-style is a bit more complicated. We can provide other means of making that easier, though (helper functions, or `Base*Codec` classes).

Reply:
to be clear, even if there are performance differences between "do 1" and "do N", I think we should consider hiding that behind a single "do N" api that dispatches to the efficient sub-case as needed.

Reply:
Done